engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EventRepository.java

```diff
@@ -17,4 +53,111 @@ public Event create(Event event) {
         statelessSessionFactory.getCurrentSession().insert(event);
         return event;
     }
+
+    /**
+     * Finds the events related to the provided org id in order to export them.
+     * It is the caller's responsibility to provide valid "from" and "to"
+     * filters.
+     * @param orgId the org id the events are related to.
+     * @param from the initial date to filter the dates from.
+     * @param to the final date to filter the dates from.
+     * @return a list of events that comply with the provided filters.
+     */
+    public List<Event> findEventsToExport(final String orgId, final LocalDate from, final LocalDate to) {
```

Comment:
Will this method be called as part of a Kafka message processing?

Reply:
Yes, once the Cloud Event is validated to have the proper format, we fetch the events before transforming them.

Reply:
Okay then could you please use a stateless session?

Reply:
Sure! Done! Thanks for the help.